pandas/core/indexes/datetimelike.py

```diff
@@ -667,6 +668,43 @@ def _sub_nat(self):
     def _sub_period(self, other):
         return NotImplemented
 
+    def _sub_period_array(self, other):
+        """
+        Subtract one PeriodIndex from another.  This is only valid if they
+        have the same frequency.
+
+        Parameters
+        ----------
+        other : PeriodIndex
+
+        Returns
+        -------
+        result : np.ndarray
+            dtype is np.int64 if there are no NaTs in self or other,
+            otherwise np.float64
+        """
+        if not is_period_dtype(self):
+            raise TypeError("cannot subtract {dtype}-dtype to {cls}"
+                            .format(dtype=other.dtype,
+                                    cls=type(self).__name__))
+
+        if not len(self) == len(other):
+            raise ValueError("cannot add indices of unequal length")
```

Comment:
I might have missed it, but is there a test that hits this?  And should it say "subtract" instead of "add"?

Reply:
Good catches on both counts.  Will fix.

Reply:
Just pushed with fix to add-->subtract.  test_pi_sub_pi, test_pi_sub_pi_with_nat, test_pi_sub_isub_pi each go through this method, though I don't think any go through this particular line